lib/TPP/ConvertTppToXsmm.cpp

```diff
@@ -151,6 +91,103 @@ getSizesAndLeadingDimsForGemmLikeOp(RewriterBase &rewriter, OpTy opTy) {
                                 ArrayRef<int64_t>{m, n, k, lda, ldb, ldc});
 }
 
+// Examples:
+// If lower=[c], higher=[a, b, c], [c] reshaped into [1, 1, c].
+// If lower=[b, c], higher=[a, b, c], [b, c] reshaped into [1, b, c].
+// If lower=[a], higher=[a, a], [a] reshaped into [1, a].
+// If lower=[a], target=[a, b, a], [a] reshaped into [1, 1, a].
+// If lower=[], target=[a, b, c], [] reshaped into [1, 1, 1].
+static void
+computeBcastShapeInput(ArrayRef<int64_t> higherRankShape,
+                       ArrayRef<int64_t> lowerRankShape,
+                       SmallVectorImpl<int64_t> &reshapeOutputShape) {
+  // Initialize new shapes with [1] * higherRank.
+  int64_t higherRank = higherRankShape.size();
+  int64_t lowerRank = lowerRankShape.size();
+
+  reshapeOutputShape.assign(higherRank, 1);
+
+  int64_t higherRankDim;
+  int64_t lowerRankDim;
+
+  for (int64_t i = higherRank - 1, j = lowerRank - 1; i >= 0 && j >= 0;
+       i--, j--) {
+    higherRankDim = higherRankShape[i];
+    lowerRankDim = lowerRankShape[j];
+
+    if (lowerRankDim == 1 && higherRankDim > 1)
+      reshapeOutputShape[i] = 1;
+    else if ((lowerRankDim > 1 && higherRankDim == 1) ||
+             (lowerRankDim == higherRankDim))
+      reshapeOutputShape[i] = lowerRankDim;
+    else if (higherRankDim != lowerRankDim)
+      assert(false && "bCast semantics for identity op broken");
+  }
+}
+
+// Given the operand type and the output type return the broadcast
+// to use in the XSMM call.
+static xsmm::BinaryFlags getBinaryBCast(MemRefType operandType,
+                                        MemRefType outputType,
+                                        size_t operandNumber) {
+
+  enum class BCastType { NONE = 0, SCALAR, ROW, COL };
+
+  auto shapeOutput = outputType.getShape();
+  auto shapeOperand = operandType.getShape();
+  assert(shapeOutput.size() >= shapeOperand.size() &&
+         "Output rank must be >= operand rank");
+  SmallVector<int64_t> bOperandShape;
+  computeBcastShapeInput(shapeOutput, shapeOperand, bOperandShape);
+  assert(shapeOutput.size() == bOperandShape.size());
+  assert(shapeOutput.size() == 2);
+
+  auto getBCastEnum = [](BCastType bCastType,
```

Comment:
I get it that this is code move, but lets avoid lambdas whenever possible. They're horrible to debug.

The alternative here is to swap the order of the `if/else` pattern below adding the results to local variables and then have the switch on that local variable.

Very little code will actually change here.

Reply:
As discussed, this code move is unnecessary and will be reverted.